### PR TITLE
Initial gRPC service for coin transfer, marker transfer and marker supply streaming (historical)

### PR DESCRIPTION
**Pull request by rchaing-figure**
Also upgrade to gradle 8.0 and Java 17.

### CODE REVIEW

File: common/src/main/kotlin/tech/figure/aggregate/common/db/DBJdbc.kt [hunk 1]
```diff
@@ -21,6 +25,54 @@ abstract class DBJdbc {
...
 
+    fun streamCoinTransfer(blockHeight: Long, denomList: List<String>): List<TxCoinTransferData> {
+        val queryStmt: String = if(denomList.isEmpty()) {
+            "SELECT * FROM COIN_TRANSFER " +
+            "WHERE BLOCK_HEIGHT >= $blockHeight"
+        } else {
+            val firstStmt = "SELECT * FROM COIN_TRANSFER " +
+            "WHERE BLOCK_HEIGHT >= $blockHeight AND "
+
+            val denomStmt = denomList.takeWhile { true }.joinToString { " DENOM = \'$it\'" }
```
Comment (hyperschwartz) [inline common/src/main/kotlin/tech/figure/aggregate/common/db/DBJdbc.kt:36]: Nit: This and the other functions in here shouldn't need the `.takeWhile { true }`.  It should behave the same without it

File: common/src/main/kotlin/tech/figure/aggregate/common/db/DBJdbc.kt [hunk 2]
```diff
@@ -21,6 +25,54 @@ abstract class DBJdbc {
...
+                .replace(",", " OR")
```
Comment (hyperschwartz) [inline common/src/main/kotlin/tech/figure/aggregate/common/db/DBJdbc.kt:37]: `joinToString` is pretty cool; it has a `separator` optional param that defaults to `, ` so you should be able to remove the `replace` call here and just use `joinToString(separator = " OR ")` and it'll magically make the query what you want it to look like

File: common/src/main/kotlin/tech/figure/aggregate/common/db/DBJdbc.kt [hunk 3]
```diff
@@ -88,3 +139,41 @@ fun List<MutableMap<String, Any>>.toTxFeeData() =
             result["SENDER"].toString()
         )
     }
+
+fun List<MutableMap<String, Any>>.toTxMakerSupply() =
```
Comment (hyperschwartz) [inline common/src/main/kotlin/tech/figure/aggregate/common/db/DBJdbc.kt:143]: Nit: `Maker` -> `Marker`

File: .github/workflows/build-publish.yml [hunk 4]
```diff
@@ -295,11 +296,11 @@ jobs:
...
           echo "VERSION=$VERSION" >> $GITHUB_ENV
-      - name: Set up JDK 11
-        uses: actions/setup-java@v1
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
         with:
-          java-version: 11
-          server-id: github
+          java-version: 17
+          distribution: 'zulu'
```
Comment (hyperschwartz) [inline .github/workflows/build-publish.yml:303]: I _think_ you might need to add another thing in here to run a `./gradlew publish` to get those protos out there in an artifact somewhere.  Maybe i'm missing something in here tho that'll just do that

File: common/src/main/kotlin/tech/figure/aggregate/common/db/DBJdbc.kt (hunk 3 above)
Comment (rchaing-figure) [reply common/src/main/kotlin/tech/figure/aggregate/common/db/DBJdbc.kt:143]: 🤦

File: proto/src/main/proto/transfer-service.proto [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  rpc CoinTransferStream(CoinTransferRequest) returns (stream CoinTransferResponse) { }
+
+  rpc MarkerSupplyStream(MarkerSupplyRequest) returns (stream MarkerSupplyResponse) { }
+
+  rpc MarkerTransferStream(MarkerTransferRequest) returns (stream MarkerTransferResponse) { }
+}
+
+message MarkerSupplyRequest {
+  int64 block_height = 1;
+  repeated string denom = 2;
```
[the remark this answers is not in the document]
Comment (rchaing-figure) [reply proto/src/main/proto/transfer-service.proto:17]: oh thats neat!

I purposely held off on comments until we got the receiving end on the TA side cause the ever changing requirements, history has shown once you put in comments, its engraved in stone.

**Review by mtps — DISMISSED** [review]

File: proto/src/main/proto/transfer-service.proto (hunk 5 above)
[the remark this answers is not in the document]
Comment (hyperschwartz) [reply proto/src/main/proto/transfer-service.proto]: Truestory + agreemented

**Review by hyperschwartz — APPROVED** [review]
Apporved pending proto package name-o's

**Pull request merged**